src/qibolab/serialize.py

```diff
@@ -79,7 +80,52 @@ def load_qubits(
     return qubits, couplers, pairs
 
 
-# This creates the compiler error
+def _load_pulse(pulse_kwargs, qubit):
+    pulse_type = pulse_kwargs.pop("type")
+    if "coupler" in pulse_kwargs:
+        q = pulse_kwargs.pop("coupler", qubit.name)
+    else:
+        q = pulse_kwargs.pop("qubit", qubit.name)
+
+    if pulse_type == "dl":
+        return Delay(**pulse_kwargs)
+    if pulse_type == "virtual_z":
+        return VirtualZ(**pulse_kwargs, qubit=q)
+    return Pulse(**pulse_kwargs, type=pulse_type, qubit=q)
+
+
+def _load_single_qubit_natives(qubit, gates) -> SingleQubitNatives:
+    """Parse native gates of the qubit from the runcard.
+
+    Args:
+        qubit (:class:`qibolab.qubits.Qubit`): Qubit object that the
+            native gates are acting on.
+        gates (dict): Dictionary with native gate pulse parameters as loaded
+            from the runcard.
+    """
+    return SingleQubitNatives(
+        **{name: _load_pulse(kwargs, qubit) for name, kwargs in gates.items()}
+    )
+
+
+def _load_two_qubit_natives(qubits, couplers, gates) -> TwoQubitNatives:
+    sequences = {}
+    for name, seq_kwargs in gates.items():
+        if isinstance(seq_kwargs, dict):
+            seq_kwargs = [seq_kwargs]
```

Comment:
I would avoid special cases, and directly serialize also lenght-1 sequences as sequences (instead of wrapping during the load process).